Source/control.cpp

```diff
@@ -592,7 +592,7 @@ void DrawSpell(const Surface &out)
 	// BUGFIX: Move the next line into the if statement to avoid OOB (SPL_INVALID is -1) (fixed)
 	if (st == RSPLTYPE_SPELL && spl != SPL_INVALID) {
 		int tlvl = myPlayer._pISplLvlAdd + myPlayer._pSplLvl[spl];
-		if (!CheckSpell(MyPlayerId, spl, st, true))
+		if (SpellFailReason::Success != CheckSpell(MyPlayerId, spl, st, true))

```

Comment:
Are we using the yoda checks @AJenbo ?

Reply:
sorry, I am a bit corrupted by MISRA standard rules

Reply:
I was initiallys sold on the idea, but i have since found that good linters do just as good a job and doesn't hurt readability.